contracts/mocks/ForeignAMBWithGasTokenMock.sol

```diff
@@ -0,0 +1,18 @@
+pragma solidity 0.4.24;
+
+import "../upgradeable_contracts/arbitrary_message/ForeignAMBWithGasToken.sol";
+
+/**
+* @title ForeignAMBWithGasTokenMock
+* @dev Wrapper on ForeignAMB contract, which supports minting gas tokens while passing messages
+*/
+contract ForeignAMBWithGasTokenMock is ForeignAMBWithGasToken {
+    function gasToken() public pure returns (IGasToken) {
+        // Address generated in unit test, also hardcoded in GasTokenMock
+        return IGasToken(0xEC8bE1A5630364292E56D01129E8ee8A9578d7D8);
```

Comment:
Is it possible to assign `0x0000000000b3F879cb30FE243b4Dfee438691c04` to the gas token contract in the ganache configuration?

Reply:
Seems like, in a current ganache version there is no easy way to do it. There is an open issue for such functionality: https://github.com/trufflesuite/ganache-cli/issues/715.

However, I was able to find a particular address and nonce used to deploy this contract. So now I can deploy a contract mock at `0x0000000000b3F879cb30FE243b4Dfee438691c04` in ganache even if I don't know the private key for deployer address.
For our case, I think, it is enough to use a slightly modified contract mock at a different address.

Reply:
OK. It is up to you. Just thought that it makes sense to reduce number of changes in the future that would be required if the change the original contract (`ForeignAMBWithGasToken`).

Reply:
For now, I will keep it as it is.
Later, when this functionality appears in ganache, it will make sense to refactor tests. So that we can use the hardcoded addresses from the mainnet, instead of mocking it every time. 

There are also similar mocks related to other functionality (sai/dai, chai). It make sense to consider refactoring them as well.